File: Unidad 3/Java/Programa21U3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.programa21u3;

import java.util.Scanner;
public class Programa21U3 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese la cantidad de números a procesar: ");
        int cantidadNumeros = sc.nextInt();

        int contador = 1, cubo;
        while (contador <= cantidadNumeros) {
            System.out.print("Ingrese un número natural positivo: ");
            int numero = sc.nextInt();
            
            if (numero > 0) {
                cubo = numero * numero * numero;
                System.out.println("El cubo de " + numero + " es " + cubo);
            } else {
                System.out.println("Por favor, ingrese un número natural positivo.");
            }
            
            contador++;
        }
    }
}

File: Unidad 3/C#/Programa21U3.cs
// See https://aka.ms/new-console-template for more information

using System;

namespace MyApp // Note: actual namespace depends on the project name.
{
    internal class Programa21U3
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Ingrese la cantidad de números a procesar: ");
            int cantidadNumeros = Convert.ToInt32(Console.ReadLine());

            int contador = 1, cubo;
            while (contador <= cantidadNumeros)
            {
                Console.WriteLine("Ingrese un número natural positivo: ");
                int numero = Convert.ToInt32(Console.ReadLine());

                if (numero > 0)
                {
                    cubo = numero * numero * numero;
                    Console.WriteLine("El cubo de " + numero + " es " + cubo);
                }
                else
                {
                    Console.WriteLine("Por favor, ingrese un número natural positivo.");
                }

                contador++;
            }
        }
    }
}